src/sagemaker/inference_recommender/inference_recommender_mixin.py

```diff
@@ -346,26 +340,66 @@ def _update_params_for_recommendation_id(
         recommendation_job_name = inference_recommendation_id.split("/")[0]
 
         sage_client = self.sagemaker_session.sagemaker_client
-        recommendation_res = sage_client.describe_inference_recommendations_job(
-            JobName=recommendation_job_name
-        )
-        input_config = recommendation_res["InputConfig"]
 
-        recommendation = next(
-            (
-                rec
-                for rec in recommendation_res["InferenceRecommendations"]
-                if rec["RecommendationId"] == inference_recommendation_id
-            ),
-            None,
-        )
+        # Retrieve model or inference recommendation job details
+        recommendation_res, model_res = None, None
+        try:
+            recommendation_res = sage_client.describe_inference_recommendations_job(
+                JobName=recommendation_job_name
+            )
+        except sage_client.exceptions.ResourceNotFound:
+            pass
+        try:
+            model_res = sage_client.describe_model(ModelName=recommendation_job_name)
```

Comment:
we don't need to make this call if `describe_inference_recommendations_job` works right?

May be we can create three private methods to modularize the code since this method is getting big.

```
def _get_model_recommendation_id(self):
   # get recommendation id from model

def _get_right_size_recommendation_id(self):
  # get recommendation id from right size job

def _get_recommendation_id(self):
  try:
    _get_right_size_recommendation_id(self)
  except:
    _get_model_recommendation_id(self)

def _update_params_for_recommendation_id(...):
  try:
    recommendation_id = _get_recommendation_id(self)
  except:
    raise ValueError("Inference Recommendation id is not valid")
```

Reply:
1. as we discussed before, what if customer's `JobName` and `ModelName` are the same, eg `thanos/abcd1234` and `thanos/abcd5678`, then we need to find out it's inference or model recommendation.   So go through two lists(inference_recommendations and model_recommendations) to get the recommendation is needed. So here I am thinking we need to loop through two lists anyway
2. good point, will create separate methods